frontend/src/libs/components/protected-route/protected-route.tsx

```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { Outlet } from 'react-router-dom';
+
+import { AppRoute } from '~/libs/enums/app-route.enum';
+import { useAppSelector, useNavigate } from '~/libs/hooks/hooks.js';
+
+const ProtectedRoute = (): React.ReactElement => {
+  const navigate = useNavigate();
+  const { users } = useAppSelector((user) => user);
+  const hasUser = Boolean(users);
```

Comment:
1. this check won't work cause `users` is an array
2. you should take authorized user from `auth` slice and check his existence